crud/room_utils.py

```diff
@@ -0,0 +1,257 @@
+        raise HTTPException(
+            status_code=404,
+            detail=f"No room type found with id {room_type_id}",
+        )
+    return _room_type
+
+
+def create_room_type(db: Session, room_type: RoomTypeCreate):
+    """Create new room type."""
+    _room_type = RoomType(
+        name=room_type.name, capacity=room_type.capacity, price=room_type.price
+    )
+    db.add(_room_type)
+    db.commit()
+    db.refresh(_room_type)
+    return _room_type
+
+
+def update_room_type(
+    db: Session, room_type_id: int, room_type: RoomTypeCreate
+):
+    """Update an existing room type."""
+    _room_type = get_room_type(db=db, room_type_id=room_type_id)
+    if not _room_type:
+        raise HTTPException(
+            status_code=404,
+            detail=f"No room type found with id {room_type_id}",
+        )
+    _room_type.name = room_type.name
+    _room_type.capacity = room_type.capacity
+    _room_type.price = room_type.price
+    db.commit()
+    db.refresh(_room_type)
+    return _room_type
+
+
+def delete_room_type(db: Session, room_type_id: int):
+    """Remove an existing room type."""
+    _room_type = get_room_type(db=db, room_type_id=room_type_id)
+    if not _room_type:
+        raise HTTPException(
+            status_code=404,
+            detail=f"No room type found with id {room_type_id}",
+        )
+    db.delete(_room_type)
+    db.commit()
+    return _room_type
+
+
+# Room features CRUD
+
+
+def get_features(db: Session, skip: int = 0, limit: int = 100):
+    """Get all features."""
+    return db.query(Feature).offset(skip).limit(limit).all()
+
+
+def get_feature(db: Session, feature_id: int):
+    """Get feature by id."""
+    _feature = db.query(Feature).filter(Feature.id == feature_id).first()
+    if not _feature:
+        raise HTTPException(
+            status_code=404, detail=f"No feature found with id {feature_id}"
+        )
+
+
+def create_feature(db: Session, feature: FeatureCreate):
+    """Create new feature."""
+    _feature = Feature(name=feature.name)
+    db.add(_feature)
+    db.commit()
+    db.refresh(_feature)
+    return _feature
+
+
+def update_feature(db: Session, feature_id: int, feature: FeatureCreate):
+    """Update an existing feature."""
+    _feature = get_feature(db=db, feature_id=feature_id)
+    if not _feature:
+        raise HTTPException(
+            status_code=404, detail=f"No feature found with id {feature_id}"
+        )
+    _feature.name = feature.name
+    db.commit()
+    db.refresh(_feature)
+    return _feature
+
+
+def delete_feature(db: Session, feature_id: int):
+    """Delete an existing feature."""
+    _feature = get_feature(db=db, feature_id=feature_id)
+    if not _feature:
+        raise HTTPException(
+            status_code=404, detail=f"No feature found with id {feature_id}"
+        )
+    db.delete(_feature)
+    db.commit()
+    return _feature
+
+
+# Facility CRUD
+
+
+def get_facilities(db: Session, skip: int = 0, limit: int = 100):
```

Comment:
What is facility here?

Reply:
Pretty much building/housing/department where the room is located. Decided to use facility as a general word.